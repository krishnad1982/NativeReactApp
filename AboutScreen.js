import React, { Component } from 'react';
import { Container, Header, Content, H1, H2, H3, Text } from 'native-base'
import { HeaderContent } from './HeaderContent';
import styles from './Assets/Styles/styles';

export class AboutScreen extends Component {
    render() {
        return (
            <Container>
                <HeaderContent title="About" {...this.props}/>
                <Content contentContainerStyle={styles.container}>
                    <Text style={styles.textSizeFive}> Since 1922 when the first Australian standards and technical documents were created to assist with the successful building of Sydney's iconic Harbor Bridge, our experts have helped companies transform the way they manage, approach and leverage risk.
        Today's world is vastly different, and far more complex. Globalization, digitization and advances in technology have dramatically changed the way companies operate as well as the sheer velocity and number of risks faced. More than ever before, these risks have the ability to severely impact business operations, resilience and brand reputation
        Managing risk proactively is now a strategic imperative, and taking an integrated approach to managing risk and compliance has never been more important.
        We help companies take a more integrated approach to managing risk. Our world-class solutions and renowned team of experts provide advice at every step ensuring companies have the information they need to make the decisions required to protect and grow their businesses and their reputation.
        We have global reach with locations across Europe, the Middle East, Africa, the Americas, Asia and the Pacific, powered by local expertise and know how. </Text>
                </Content>
            </Container>
        );
    }
}