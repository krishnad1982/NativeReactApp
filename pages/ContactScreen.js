import React, { Component } from 'react';
import { View, StyleSheet, Text } from 'react-native';
import { Container, Content, Input, Item, Textarea, Form, Label, Picker, Icon, Button } from 'native-base'
import { HeaderContent } from '../utilities/HeaderContent';

export class ContactScreen extends Component {
    constructor(props) {
        super(props);
        this.state = {
            selected: undefined
        };
    }
    onValueChange(value: string) {
        this.setState({
            selected: value
        });
    }
    render() {
        return (
            <Container>
                <HeaderContent title="Contact" {...this.props} />
                <Content contentContainerStyle={{ padding: 10 }}>
                    
                        <Item floatingLabel>
                            <Label>Name</Label>
                            <Input />
                        </Item>
                        <Item floatingLabel>
                            <Label>Email</Label>
                            <Input />
                        </Item>
                        <Item picker>
                            <Picker
                                mode="dropdown"
                                iosHeader="Select your SIM"
                                iosIcon={<Icon name="ios-arrow-down-outline" />}
                                style={{ width: undefined }}
                                selectedValue={this.state.selected}
                                onValueChange={this.onValueChange.bind(this)}
                            >
                                <Picker.Item label="Male" value="Male" />
                                <Picker.Item label="Female" value="Female" />
                            </Picker>
                        </Item>
                        <Textarea rowSpan={5} bordered placeholder="Message" />
                        <Button block success><Text> Submit </Text></Button>
                        <Button block warning><Text> Cancel </Text></Button>
                   
                </Content>
            </Container>
        );
    }
}