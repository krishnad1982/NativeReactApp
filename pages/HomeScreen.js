import React, { Component } from 'react';
import { AppRegistry, ActivityIndicator, Image, Text, View, FlatList, ScrollView, StyleSheet } from 'react-native';
import { Card } from "react-native-elements";
import { Container, Content, Header, Left, Body, Right, Button, Icon, Title } from 'native-base'
import styles from '../Assets/styles/styles';
import { HeaderContent } from '../utilities/HeaderContent';

export class HomeScreen extends Component {
    constructor(props) {
        super(props);
        this.state = { isLoading: true }
    }

    componentDidMount() {
        return fetch('http://newsapi.org/v2/top-headlines?country=us&category=business&apiKey=364a811e3dfc495b9cdfb4f4a3e70112')
            .then((response) => response.json())
            .then((responsejson) => {
                this.setState({
                    isLoading: false,
                    dataSource: responsejson.articles,
                }, function () {

                });
            })
            .catch((error) => {
                console.log(error);
            })
    }

    render() {
        if (this.state.isLoading) {
            return (
                <Container>
                    <HeaderContent title="Home" {...this.props} />
                    <Content contentContainerStyle={{ flex: 1, padding: 20 }}>
                        <ActivityIndicator />
                    </Content>
                </Container>
            )
        }

        return (
            <Container>
                <HeaderContent title="Home" {...this.props} />
                <Content contentContainerStyle={{ flex: 1, paddingTop: 20, shadowOpacity: 0.75, shadowRadius: 5, shadowOffset: { height: 0, width: 0 }, shadowColor: 'red' }}>
                    <FlatList
                        data={this.state.dataSource}
                        renderItem={({ item }) =>
                            <Card title={item.title} image={{ uri: item.urlToImage }} containerStyle={{ padding: 0, width: 380 }}>
                                <Text style={{ marginBottom: 5 }}>{item.author}</Text>
                                <Text style={{ marginBottom: 10 }}>{item.publishedAt}</Text>
                            </Card>
                        }
                        keyExtractor={item => item.title}
                    />
                </Content>
            </Container>
        );
    }
}