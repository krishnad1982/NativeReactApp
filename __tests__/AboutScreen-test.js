import React from 'react';
import { AboutScreen } from '../pages/AboutScreen';
import renderer from 'react-test-renderer';


test('renders correctly', () => {
    const tree = renderer.create(<AboutScreen />).toJSON();
    expect(tree).toMatchSnapshot();
});