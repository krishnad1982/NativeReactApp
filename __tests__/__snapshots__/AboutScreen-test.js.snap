// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F8F8F8",
            "borderBottomColor": "#a7a6ab",
            "borderBottomWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 6,
            "paddingRight": 10,
            "paddingTop": 18,
            "right": 0,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "top": 0,
          },
          Object {
            "backgroundColor": "#B80000",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginLeft": -3,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 4,
              "paddingRight": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#007aff",
                    "fontSize": 34,
                    "marginLeft": 1,
                    "marginRight": 2,
                    "marginTop": 0,
                    "paddingTop": 1,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "700",
              "marginLeft": undefined,
              "paddingLeft": 4,
              "paddingTop": 1,
              "textAlign": "center",
            }
          }
        >
          About
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 50,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginRight": -9,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#007aff",
                    "fontSize": 24,
                    "marginLeft": 2,
                    "marginRight": 0,
                    "marginTop": 0,
                    "paddingTop": 2,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        Object {
          "justifyContent": "center",
          "marginTop": 2,
          "padding": 10,
        },
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    viewIsInsideTabBar={false}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "fontSize": 15,
            },
          ]
        }
        uppercase={false}
      >
         Since 1922 when the first Australian standards and technical documents were created to assist with the successful building of Sydney's iconic Harbor Bridge, our experts have helped companies transform the way they manage, approach and leverage risk. Today's world is vastly different, and far more complex. Globalization, digitization and advances in technology have dramatically changed the way companies operate as well as the sheer velocity and number of risks faced. More than ever before, these risks have the ability to severely impact business operations, resilience and brand reputation Managing risk proactively is now a strategic imperative, and taking an integrated approach to managing risk and compliance has never been more important. We help companies take a more integrated approach to managing risk. Our world-class solutions and renowned team of experts provide advice at every step ensuring companies have the information they need to make the decisions required to protect and grow their businesses and their reputation. We have global reach with locations across Europe, the Middle East, Africa, the Americas, Asia and the Pacific, powered by local expertise and know how. 
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
